### PR TITLE
Story Editor: Added List Text Sets

### PR DESCRIPTION
**Pull request by maxyinger**
## Summary
Adds text sets under `List 1` category in this design file:
https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=837%3A3585

## User-facing changes
NA

## Testing Instructions
Enable the text sets flag in experiments and see the new text sets in the story editor.

---

Partial for #4078

### CODE REVIEW

**Review by miina — CHANGES_REQUESTED** [review]
Similar comments as in other PR-s, otherwise looks good!
- There are 3 lists (the first three) where only the font differs, only one is currently implemented:
Screenshot 2020-09-21 at 11 53 33

- There is one light line that differs from others (Is this the dotted lines set? Would be good to confirm with Sam if we should skip this for now or opt for using solid lines):
Screenshot 2020-09-21 at 11 55 00

- The margins between the title and the content differ a bit in some cases, for example:
Screenshot 2020-09-21 at 11 58 28

- The `+` marks are a bit off in terms of vertical centering based on the text, for example:
Screenshot 2020-09-21 at 12 00 52

- The `-` markers look a bit different, e.g. the second last looks thinner compared to the previous ones:
Screenshot 2020-09-21 at 12 01 59

**Comment by miina** [comment]
One more note: List 1 and List 2 are separately in Figma but these would still be one category AFAIK and would be good to keep it in one file under just "list".

**Comment by maxyinger** [comment]
@miina 

> There are 3 lists (the first three) where only the font differs, only one is currently implemented:

Yea so the first three of every set use roboto, then the iOS & Windows system font which we don't currently have access to so I've just implemented the roboto set. (I'm not sure how one even sets a font to the system font in the editor rn)

> There is one light line that differs from others (Is this the dotted lines set? Would be good to confirm with Sam if we should skip this for now or opt for using solid lines):

Awesome. yup can't do dotted lines in editor so just used solid lines. will confirm it's cool with sam.

As for lines looking different, I think this is just an issue of rendering subpixel lines in the editor. They're all set to 1px (CSS px) wide in story space, but I think when that gets interpreted to actual screen space it can be less than 1 CSS pixel which the browser doesn't consistently render. This is a good case for building some notion of borders/outlines/stroke in stories that maintain line width regardless of element scale.

> The margins between the title and the content differ a bit in some cases, for example:

Yea this is the same issue that was brought up in the other PR. I'll go through these again with whatever approach we settle on here, let me know if you've decided on a certain approach or want to discuss it more.

> The + marks are a bit off in terms of vertical centering based on the text, for example:

Yea since these are text and the `+` is not vertically centered relative to it's bounding box, there's no way to automate centering these and had to do it manually. Can try and tighten these up some more

> The - markers look a bit different, e.g. the second last looks thinner compared to the previous ones:

Yeaup. this is that same subpixel issue mentioned above.

File: assets/src/edit-story/components/library/panes/text/textSets/loadTextSets.js
Comment (miina) [inline assets/src/edit-story/components/library/panes/text/textSets/loadTextSets.js]: Could we rename this to just `list`? All the lists would be under the `list` category, there won't be a separate list 1 and list 2 category, as far as I know.

File: assets/src/edit-story/components/library/panes/text/textSets/raw/list_1.json
Comment (miina) [inline assets/src/edit-story/components/library/panes/text/textSets/raw/list_1.json]: ^ This as well, we could rename to just `list.json`.

Comment (maxyinger) [reply assets/src/edit-story/components/library/panes/text/textSets/raw/list_1.json]: Cool, I'm just gonna add the list 2 sets to this PR too then

**Review by dmmulroy — APPROVED** [review]

**Review by miina — COMMENTED** [review]
Looks great! Just two details found:

1. In this case the heading is "Regular" font-weight in Figma, it seems to be "light" in the editor. About the checkboxes -- the box that is checked is wider than the others. Also, it's all black shapes instead of what's in Figma but I assume that's a workaround until we have it available?
Screenshot 2020-09-30 at 11 47 43

2. In this text set, currently, the background is set to solid gray color, however, in Figma, it's black with 0.05 opacity -- we could also do the same here -- use black with 0.05 opacity -- might look better in the preview, too!
Screenshot 2020-09-30 at 11 58 09

**Comment by maxyinger** [comment]
> Looks great! Just two details found:
> 
> 1. In this case the heading is "Regular" font-weight in Figma, it seems to be "light" in the editor. About the checkboxes -- the box that is checked is wider than the others. Also, it's all black shapes instead of what's in Figma but I assume that's a workaround until we have it available?
> 
> Screenshot 2020-09-30 at 11 47 43
> 
> 1. In this text set, currently, the background is set to solid gray color, however, in Figma, it's black with 0.05 opacity -- we could also do the same here -- use black with 0.05 opacity -- might look better in the preview, too!
> 
> Screenshot 2020-09-30 at 11 58 09

Ah good catches. And yea, those are the temporary styles for the checkboxes sam recommended so far

**Comment by maxyinger** [comment]
Yea, opacity on the text box fill made the preview look much nicer. good call:
Screen Shot 2020-09-30 at 3 59 21 PM

**Review by miina — APPROVED** [review]
Thanks for the update, looks much better now!

On the list with checkboxes noticed one more detail (sorry, trying to get them as close to perfect as possible!):

It looks like the checkboxes aren't centered compared to the text, they seem to be lower than the text:
Screenshot 2020-10-01 at 09 49 03

If we could move each text 1-2px lower (probably the easiest to do from the design panel Y position input directly) / the checkboxes 1-2px higher, would be great. Tried out locally with 1px and it looks better, perhaps that'd be enough:
Screenshot 2020-10-01 at 09 48 20

(Also, the heading "The principles" still seems to have "Light" as the font-weight although it's regular in Figma.)

Will approve pending these two details so you don't have to wait for my approval tomorrow (timezone difference).

**Review by BrittanyIRL — APPROVED** [review]
Good work on these. 
2 minor size discrepancies i spotted: 

the nunito headline should probably round to 31px not 30px
Screen Shot 2020-10-05 at 11 19 03 AM
Screen Shot 2020-10-05 at 11 18 58 AM

And then the titillium web headline should be 27px not 26px.
Screen Shot 2020-10-05 at 11 18 28 AM


Screen Shot 2020-10-05 at 11 18 23 AM

**Pull request merged**
